dolphin-dnn/src/main/java/edu/snu/dolphin/dnn/blas/Matrix.java

```diff
@@ -0,0 +1,396 @@
+/*
+ * Copyright (C) 2015 Seoul National University
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package edu.snu.dolphin.dnn.blas;
+
+/**
+ * Matrix interface.
+ */
+public interface Matrix {
+
+  /**
+   * Returns the number of rows.
+   */
+  int getRows();
+
+  /**
+   * Returns the number of columns.
+   */
+  int getColumns();
+
+  /**
+   * Returns a element specified by the given index (linear indexing).
```

Comment:
Please include comments on how this 'linear indexing' is processed: row-wise or column-wise?